### ISSUE TITLE
Support pre-existing mipmaps(and only one miplevel)

### ISSUE THREAD

**Issue by progsung** [CesiumGS/cesium-unity#274]
Hi, I have a opinion about mipmap process #246.

In my opinion to "Support pre-existing mipmaps", need to check miplevel count more than one. 
**If it has only one miplevel, need to generate mipmap like before.**

Also, I reviewed part of KTX2 spec about mipmip levels.
> levelCount=1
  means that a file contains only the base level and the texture isn’t meant to have other levels...
https://registry.khronos.org/KTX/specs/2.0/ktxspec.v2.html#_levelcount

Thanks.

**Comment by tomlee-dev** [comment]
I created a PR.

https://github.com/CesiumGS/cesium-unity/pull/284

please check this.

Thanks!

**Comment by kring** [comment]
Thanks for the PR @tomlee-dev! And for the issue @progsung!

 I think that this logic is still not quite right, though.

As mentioned above, in KTX2 there is a distinction made between "this image has no mipmaps, so they should be generated at runtime" and "this image has no mipmaps because it makes no sense to create a mipmap for this type of image." It is, confusingly, encoded in the `levelCount` property. With `levelCount=0`, mipmaps should be generated. With `levelCount=1`, mipmaps make no sense. This distinction corresponds, I think, to whether `mipPositions` is empty. An empty `mipPositions` means to go ahead and generate them. A single `mipPosition` means "don't generate any more."  In other words, the logic here in cesium-unity is already correct and needs no changes.

However, cesium-native is not behaving correctly. It loses this distinction, and will always fill in `mipPositions` for KTX2 images. So cesium-native should be modified to only fill in mipPositions if `levelCount` is greater than 0.

However, the KTX2 API that cesium-native is using doesn't expose levelCount at all. Instead, it exposes a `numLevels` property. That property sounds the same, but it's different in that it will _never_ be 0. Both of our cases above will have a `numLevels` of 1.  But, when `levelCount` is 0 in the file, the KTX2 reader will also set the `generateMipmaps` property to true.

So, in summary, in `GltfReader.cpp`, cesium-native should only populate `mipPositions` when `pTexture->generateMipmaps` is false.

**Comment by progsung** [comment]
Thanks. I will keep on eyes about it.
